plugin-hrm-form/src/services/fetchApi.ts

```diff
@@ -75,6 +75,10 @@ export const fetchApi = async (baseUrl: URL, endpointPath: string, options: Requ
     }
     throw new ApiError(`Error response: ${response.status} (${response.statusText})`, { response, body });
   }
-
-  return response.json() as Promise<any>;
+  let result;
```

Comment:
This approach means that we parse all responses as json, and if it fails for any reason, it swallows the exception and returns `undefined`? I don't think this is the best strategy at such a low level in our API. It will hide issues with malformed JSON where the response is supposed to parsable as JSON, making it throw a harder to diagnose error elsewhere. It will also clobber any plaintext responses where we actually care about the content.

If a response isn't shouldn't be treated as JSON, it should be reflected in it's `Content-Type` header, and having experimented, the `DELETE` endpoint does indeed return `text/plain` as a content type. So instead of swallowing errors I would suggest something like this:

```suggestion
  if ((response.headers.get('Content-Type') ?? '').toLowerCase().includes('json')) {
    return response.json();
  }
  return response.text();
```

So we are a bit smarter than previously about treating JSON as JSON and text as text - this way we till get parser errors if something is _supposed_ to be JSON but isn't, and we get the content of plain text.

(I did a brief test of the suggested change above and it seems to work btw)

Reply:
Thank you!